python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -146,20 +149,55 @@ def get_hashable(self) -> Hashable:
         schema_hash = tuple(self.schema.items())
         return (type(self), schema_hash, args)
 
+    # Hacky to avoid type-checking issues, just advertise the
+    # signature. Both mypy and pyright complain if we have an abstract
+    # method that takes arbitrary *args, but the subclasses have
+    # tighter signatures. This complaint is correct because the
+    # subclass is not Liskov-substitutable for the superclass.
+    # However, we know do_evaluate will only be called with the
+    # correct arguments by "construction".
+    do_evaluate: Callable[..., DataFrame]
+    """
+    Evaluate the node (given its evaluated children), and return a dataframe.
+
+    Parameters
+    ----------
+    args
+        Non child arguments followed by any evaluated dataframe inputs.
+
+    Returns
+    -------
+    DataFrame (on device) representing the evaluation of this plan
+    node.
+
+    Raises
+    ------
+    NotImplementedError
+        If we couldn't evaluate things. Ideally this should not occur,
+        since the translation phase should pick up things that we
+        cannot handle.
```

Comment:
Let's avoid the "we" here.
```suggestion
        If evaluation fails. Ideally this should not occur, since the
        translation phase should fail earlier.
```